roles/ceph-update/tasks/main.yml

```diff
@@ -8,41 +8,78 @@
   delay: 60
   delegate_to: "{{ groups['ceph_monitors'][0] }}"
 
-- name: restart ceph mons
-  service: name=ceph-mon-all state=restarted
+- name: stop ceph mon
+  service: name=ceph-mon-all state=stopped
+  when: "'ceph_monitors' in group_names"
+
+- name: correct ceph dir owner
+  file: path=/var/lib/ceph state=directory owner=ceph group=ceph recurse=yes
+  when: upgrade_ceph2jewel|bool and "ceph_monitors" in group_names
+
+- name: start ceph mons
+  service: name=ceph-mon-all state=started enabled=yes
+  when: "'ceph_monitors' in group_names"
 
 # Setting noout to prevent cluster from rebalancing after service restart
 - name: set osd noout
   command: ceph osd set noout
   delegate_to: "{{ groups['ceph_monitors'][0] }}"
-  when: inventory_hostname == groups['ceph_osds'][0]
+  when: "'ceph_osds' in group_names"
 
-# Setting osd values for first osd to reduce performance degradation of client I/O
+# Setting osd values to reduce performance degradation of client I/O
 - name: set osd noscrub
   command: ceph osd set noscrub
   delegate_to: "{{ groups['ceph_monitors'][0] }}"
-  when: inventory_hostname == groups['ceph_osds'][0]
+  when: "'ceph_osds' in group_names"
 
 - name: set osd nodeep scrub
   command: ceph osd set nodeep-scrub
   delegate_to: "{{ groups['ceph_monitors'][0] }}"
-  when: inventory_hostname == groups['ceph_osds'][0]
+  when: "'ceph_osds' in group_names"
+
+- name: stop ceph osd
+  service: name=ceph-osd-all state=stopped
+  when: "'ceph_osds' in group_names"
 
-- name: restart ceph osds
-  service: name=ceph-osd-all state=restarted
+- name: correct owner of ceph dir and journal partitions
+  include: ceph_owner.yml
+  when: upgrade_ceph2jewel|bool and 'ceph_osds' in group_names
```

Comment:
there are two flags, `upgrade_ceph` means ceph upgrade,  and `upgrade_ceph2jewel` means upgrading ceph from hammer to jewel. `upgrade_ceph` trigger restart, while `upgrade_ceph2jewel` trigger checking ownership of ceph dir.